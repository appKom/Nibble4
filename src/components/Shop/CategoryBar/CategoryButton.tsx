import React, { FC } from "react";
import styled from "styled-components";
import { OnlineBlue, OnlineOrange } from "style";

type CategoryButtonProps = {
  text: string;
};

const CategoryButton: FC<CategoryButtonProps> = ({
  text,
}: CategoryButtonProps) => {
  return <Button> {text} </Button>;
};

export default CategoryButton;

const Button = styled.button`
  background-color: ${OnlineBlue};
  color: white;
  border: none;
  border-radius: 2px;
  margin: 4px;
  width: 6%;
  height: 25px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);

  :active {
    background-color: ${OnlineOrange};
  }

  :focus {
    outline: none;
  }
`;
