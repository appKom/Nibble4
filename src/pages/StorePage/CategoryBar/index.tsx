import React, { useContext, FC } from "react";
import styled from "styled-components";
import Category from "./Category";
import { StateContext } from "state/state";
import { favouritesCategory, getCategories } from "types/inventory";
import { setCategory } from "state/actions";

const CategoryBar: FC = () => {
  const { state, dispatch } = useContext(StateContext);
  const categories = getCategories(state.inventory, state.user!);

  // If the logged in user has a substantial amount of products previously purchased,
  // automatically set the category to "Dine favoritter".
  if (state.user!.favourites.length >= 4)
    dispatch(setCategory(favouritesCategory));

  return (
    <Container>
      {categories.map((category) => (
        <Category key={category} category={category} />
      ))}
    </Container>
  );
};

export default CategoryBar;

const Container = styled.div`
  display: flex;
  margin-left: 25px;
  height: 30px;
`;
