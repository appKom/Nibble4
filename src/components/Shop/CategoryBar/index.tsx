import React, { useState, useContext, FC } from "react";
import styled from "styled-components";
import CategoryButton from "./CategoryButton";
import { StateContext } from "state/state";
import { getCategories } from "types/inventory";

const CategoryBar: FC = () => {
  const { state } = useContext(StateContext);

  const categories = getCategories(state.inventory);

  const knapper = categories.map((category) => (
    <CategoryButton key={category} category={category} />
  ));

  return <Container> {knapper} </Container>;
};

export default CategoryBar;

const Container = styled.div`
  display: flex;
  margin-left: 25px;
`;
