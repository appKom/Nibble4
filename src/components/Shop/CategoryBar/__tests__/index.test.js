import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import CategoryBar from "../index";

afterEach(cleanup);
it("renders", () => {
  const { asFragment } = render(<CategoryBar />);
  expect(asFragment()).toMatchSnapshot();
});
