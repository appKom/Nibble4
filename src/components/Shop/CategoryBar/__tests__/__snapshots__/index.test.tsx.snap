// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-gsTCUz hBNOmI"
  >
     
    <button
      class="sc-bdfBwQ dRtQbr"
      style="background-color: rgb(249, 183, 89);"
    >
      Alt
    </button>
     
  </div>
</DocumentFragment>
`;
